apiserver/registration.go

```diff
@@ -172,17 +169,18 @@ func (h *registerUserHandler) processPost(
 	return userTag, response, nil
 }
 
-func getConnectorInfoer(ctx context.Context, model stateenvirons.Model, cloudService common.CloudService, credentialService common.CredentialService) (environs.ConnectorInfo, error) {
-	configGetter := stateenvirons.EnvironConfigGetter{
-		Model: model, CloudService: cloudService, CredentialService: credentialService}
-	environ, err := common.EnvironFuncForModel(model, cloudService, credentialService, configGetter)(ctx)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	if connInfo, ok := environ.(environs.ConnectorInfo); ok {
-		return connInfo, nil
+func getConnectorInfoer(
```

Comment:
This function can be completely deleted by simply just using a provider getter.

Reply:
We still need the function so we can patch it / mock it out in testing.

Reply:
I don't see this. Are you able to provide more information.

Reply:
https://github.com/juju/juju/blob/34de810d3d7fb85fc054177d51263b05700f3dfd/apiserver/registration_test.go#L95

Reply:
This can just be replaced with a mock.

Reply:
fine, but in that case, we still need to define a field on the `registerUserHandler` that provides this function and can be mocked out for testing. So this still doesn't allow us to get rid of the function.

Reply:
I fail to see any of this. We can provide the dependency in a better means to registerUserHandler by passing in the a provider getter directly on to the struct. We can mock out the ctx. There are a number of options here to make this code cleaner.

Reply:
alright, done now